database connection errors

@claude

my render deployments are all successful but I'm getting issues with actually visiting the site

1 - these are the current env variables on render: ADMIN_PASSWORD_HASH=##secret## ADMIN_USERNAME=admin ALLOWED_ORIGINS=https://manaandmeeples.co.nz,https://www.manaandmeeples.co.nz,http://localhost:3000 COOKIE_DOMAIN=.manaandmeeples.co.nz CORS_ORIGINS=https://mana-meeples-singles-market.onrender.com DATABASE_URL="postgresql://tcg_admin:1FhON1ZvCR7bRry4L9UoonvorMD4BjAR@dpg-d3i3387diees738trbg0-a/tcg_singles?sslmode=require" JWT_SECRET=##secret## NODE_VERSION=20.x OWNER_EMAIL=manaandmeeples@gmail.com PGSSLMODE=require PNPM_VERSION=9.x POKEMON_TCG_API_KEY=e76411e2-f116-47b0-b41a-2db362e32700 PORT=10000 SESSION_SECRET=##secret## SMTP_FROM=manaandmeeples@gmail.com SMTP_HOST=smtp.gmail.com SMTP_PASSWORD=mccucmcnbvgyxmwr SMTP_PORT=587 SMTP_SECURE=false SMTP_USER=manaandmeeples@gmail.com

2- C:\Git_Repos\mana_meeples_singles_market>curl -sS https://mana-meeples-api.onrender.com/api/health {"ok":true,"service":"Mana & Meeples API","version":"1.0.0","time":"2025-10-18T22:38:26.164Z"} C:\Git_Repos\mana_meeples_singles_market>curl -sS "https://mana-meeples-api.onrender.com/api/cards?limit=1" {"error":"Failed to fetch cards"}
C:\Git_Repos\mana_meeples_singles_market>curl -sSI https://mana-meeples-web.onrender.com/ HTTP/1.1 200 OK Date: Sat, 18 Oct 2025 22:31:00 GMT Content-Type: text/html; charset=utf-8 Connection: keep-alive Cache-Control: public, max-age=0, s-maxage=300 etag: W/"a58b92faa93eec6b83362b9d9fca5525" last-modified: Sat, 18 Oct 2025 22:29:43 UTC strict-transport-security: max-age=315360000; includeSubdomains; preload vary: Accept-Encoding vary: Accept-Encoding x-content-type-options: nosniff CF-Cache-Status: MISS Server: cloudflare CF-RAY: 990b7bc07e41d9ac-AKL alt-svc: h3=":443"; ma=86400

C:\Git_Repos\mana_meeples_singles_market > node scripts/smoke.mjs "https://mana-meeples-web.onrender.com" "https://mana-meeples-api.onrender.com" WARN: /api/health/db failed: https://mana-meeples-api.onrender.com/api/health/db -> HTTP 500 ✖ smoke failed: Error: https://mana-meeples-api.onrender.com/api/cards?limit=1 -> HTTP 500 at checkJSON (file:///C:/Git_Repos/mana_meeples_singles_market/scripts/smoke.mjs:14:20) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async file:///C:/Git_Repos/mana_meeples_singles_market/scripts/smoke.mjs:47:17  ELIFECYCLE  Command failed with exit code 1.

@claude

C:\Git_Repos\mana_meeples_singles_market>pnpm --filter ./apps/api run build

> @mana-meeples/api@ build C:\Git_Repos\mana_meeples_singles_market\apps\api
> tsc -p tsconfig.json

src/services/variationAnalysis.ts:328:16 - error TS2339: Property 'rows' does not exist on type 'any[]'.

328     if (result.rows.length === 0) {
                   ~~~~

src/services/variationAnalysis.ts:332:19 - error TS2339: Property 'rows' does not exist on type 'any[]'.

332     return result.rows[0];
                      ~~~~

src/services/variationAnalysis.ts:359:16 - error TS2339: Property 'rows' does not exist on type 'any[]'.

359     if (result.rows.length === 0) {
                   ~~~~

src/services/variationAnalysis.ts:363:19 - error TS2339: Property 'rows' does not exist on type 'any[]'.

363     return result.rows[0];
                      ~~~~

src/services/variationAnalysis.ts:411:18 - error TS2339: Property 'rows' does not exist on type 'any[]'.

411       if (result.rows.length > 0) {
                     ~~~~

src/services/variationAnalysis.ts:412:16 - error TS2339: Property 'rows' does not exist on type 'any[]'.

412         result.rows[0].finishes = finishResult.rows[0]?.finishes || [];
                   ~~~~

src/services/variationAnalysis.ts:412:48 - error TS2339: Property 'rows' does not exist on type 'any[]'.

412         result.rows[0].finishes = finishResult.rows[0]?.finishes || [];
                                                   ~~~~

src/services/variationAnalysis.ts:413:23 - error TS2339: Property 'rows' does not exist on type 'any[]'.

413         return result.rows[0];
                          ~~~~

src/services/variationAnalysis.ts:450:32 - error TS2339: Property 'rows' does not exist on type 'any[]'.

450         treatments: treatments.rows[0]?.treatments || [],
                                   ~~~~

src/services/variationAnalysis.ts:451:32 - error TS2339: Property 'rows' does not exist on type 'any[]'.

451         border_colors: borders.rows[0]?.border_colors || [],
                                   ~~~~

src/services/variationAnalysis.ts:452:28 - error TS2339: Property 'rows' does not exist on type 'any[]'.

452         promo_types: foils.rows[0]?.promo_types || [],
                               ~~~~

src/services/variationAnalysis.ts:453:31 - error TS2339: Property 'rows' does not exist on type 'any[]'.

453         frame_effects: frames.rows[0]?.frame_effects || [],
                                  ~~~~

src/services/variationAnalysis.ts:454:28 - error TS2339: Property 'rows' does not exist on type 'any[]'.

454         finishes: finishes.rows[0]?.finishes || []
                               ~~~~

src/services/variationAnalysis.ts:459:19 - error TS2339: Property 'rows' does not exist on type 'any[]'.

459     return result.rows.length > 0 ? result.rows[0] : null;
                      ~~~~

src/services/variationAnalysis.ts:459:44 - error TS2339: Property 'rows' does not exist on type 'any[]'.

459     return result.rows.length > 0 ? result.rows[0] : null;
                                               ~~~~


Found 15 errors in the same file, starting at: src/services/variationAnalysis.ts:328

C:\Git_Repos\mana_meeples_singles_market\apps\api:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @mana-meeples/api@ build: `tsc -p tsconfig.json`
Exit status 2

--- a/apps/api/src/services/variationAnalysis.ts
+++ b/apps/api/src/services/variationAnalysis.ts
@@ -324,12 +324,12 @@ async function getSetVariations(setId: any) {
       FROM set_variations_metadata
       WHERE set_id = $1
     `, [setId]);
-    
-    if (result.rows.length === 0) {
+
+    if (result.length === 0) {
       return null;
     }
-    
-    return result.rows[0];
+
+    return result[0];
   } catch (error) {
     console.error('Error getting set variations:', error);
     throw error;
@@ -355,12 +355,12 @@ async function getGameVariations(gameId: any) {
       FROM game_variations_metadata
       WHERE game_id = $1
     `, [gameId]);
-    
-    if (result.rows.length === 0) {
+
+    if (result.length === 0) {
       return null;
     }
-    
-    return result.rows[0];
+
+    return result[0];
   } catch (error) {
     console.error('Error getting game variations:', error);
     throw error;
@@ -408,9 +408,9 @@ async function getVariationFilters(gameId = null, setId = null) {
       `, [gameId]);
       
       const result = await db.query(query, params);
-      if (result.rows.length > 0) {
-        result.rows[0].finishes = finishResult.rows[0]?.finishes || [];
-        return result.rows[0];
+      if (result.length > 0) {
+        result[0].finishes = finishResult[0]?.finishes || [];
+        return result[0];
       }
       return null;
     } else {
@@ -447,16 +447,16 @@ async function getVariationFilters(gameId = null, setId = null) {
       ]);
       
       return {
-        treatments: treatments.rows[0]?.treatments || [],
-        border_colors: borders.rows[0]?.border_colors || [],
-        promo_types: foils.rows[0]?.promo_types || [],
-        frame_effects: frames.rows[0]?.frame_effects || [],
-        finishes: finishes.rows[0]?.finishes || []
+        treatments: treatments[0]?.treatments || [],
+        border_colors: borders[0]?.border_colors || [],
+        promo_types: foils[0]?.promo_types || [],
+        frame_effects: frames[0]?.frame_effects || [],
+        finishes: finishes[0]?.finishes || []
       };
     }
     
     const result = await db.query(query, params);
-    return result.rows.length > 0 ? result.rows[0] : null;
+    return result.length > 0 ? result[0] : null;
     
   } catch (error) {
     console.error('Error getting variation filters:', error);

